### ISSUE TITLE
404 Error

### ISSUE THREAD

**Issue by bkonia** [jpwalters/OpenVBX-Plugin-SMS-Conversation#25]
Version 1.0 worked fine for me. However, starting with version 1.1, the following error message constantly pops up:

An unknown error occurred. Please contact your OpenVBX provider. Unable to complete request: https://vbx.hostland.com/index.php/plugins/sms-conversation/plugin.json?_=1461699753719

I've confirmed that after reverting to 1.0, the error no longer occurs.

**Comment by jpwalters** [comment]
Interesting ... the call to get the plugin.json file was added in version 1.1 so that would explain why you don't see the issue in version 1.0.  BUT the file exist in both versions so I don't understand why you're getting a 404 error when it tries to retrieve that file.  
1. Can you verify that in your sms-conversation plugin folder that the file plugin.json exit?
2. When you say it constantly pops up do you mean each time you load the sms plugin page or even while you're on the page?  The plugin.json file is only coded to load once when the plugin loads so that doesn't make sense that it would keep popping up while you're on the page.

**Comment by jpwalters** [comment]
FYI ... version 1.2 has been released.  Possibly that resolves this issue?

**Comment by bkonia** [comment]
Yes, the file exists. The following URL loads the file with no error:

https://vbx.hostland.com/plugins/sms-conversations/plugin.json

However, the plugin is trying to access:

https://vbx.hostland.com/index.php/plugins/sms-conversation/plugin.json

Therefore, there must be something wrong with the router configuration for
the plugin in index.php. I'm using the latest version and it's still doing
the same thing.
ᐧ

**Comment by jpwalters** [comment]
I didn't noticed the "index.php" in the url which shouldn't be there.  I'll have to figure out why that's getting inserted.

**Comment by gegere** [comment]
Having index.php display or not display is part or the CodeIgnitor setup. Look within the config.php file and adjust as best fits your needs.

**Comment by jpwalters** [comment]
Oh right ... so that's an "OpenVBX" setting.  You can find the config.php located at OpenVBX\config\config.php.  

If you look for the below code block try setting index_page to blank (`$config['index_page'] = '';`) and see if the issue is resolved.

```
/*
|--------------------------------------------------------------------------
| Index File
|--------------------------------------------------------------------------
|
| Typically this will be your index.php file, unless you've renamed it to
| something else. If you are using mod_rewrite to remove the page set this
| variable so that it is blank.
|
*/
$config['index_page'] = 'index.php';
```

**Comment by bkonia** [comment]
Since I'm not using mod_rewrite, index.php is correct and that's how it's
configured in my config.php. All other links within OpenVBX work fine. It
only seems to be a problem with this plugin.
ᐧ

**Comment by jpwalters** [comment]
In version 1.0 we found an issue that if you install OpenVBX in a subdirectory the URL was being incorrectly translated.  And therefore we made the following one line change.

Link to Commit: https://github.com/jpwalters/OpenVBX-Plugin-SMS-Conversation/commit/4a648a67af9be25833023ae9283f6437508a0422#diff-4f8f9ea801b6c15533c04fe8fb2751a7

OLD: `$.post( "/messages/sms/" ...`
NEW: `$.post(OpenVBX.home + "/messages/sms/" ...`

`OpenVBX.home` is a global javascript variable provided by the OpenVBX framework which holds the path to the root or home directory where OpenVBX has been installed.  For me that variable never includes index.php so its really odd that it does for you.

Can you post the contents of your .htaccess file?  Maybe seeing that will help me reproduce the issue.

**Comment by jpwalters** [comment]
Looking at the code I see a way to work around the issue.  Most likely to your point it is a routing issues but all routing is handled in the OpenVBX framework so to get that changed we'd have to find the issues, likely fix it ourselves and then encourage the OpenVBX project owner to include the fix into the main project.  Therefore I'll likely just move toward my own work around solution that keeps the solution within this plugin.  

It would still be interesting to understand why index.php is getting including in your installation when it doesn't in mine.

**Comment by gegere** [comment]
Many more good details about this from a pull request I created https://github.com/jpwalters/OpenVBX-Plugin-SMS-Conversation/pull/6 

Happy to help with more code bits.

**Comment by bkonia** [comment]
Hi James,

In OpenVBX->Settings->System Config, there's an option, "Do you want to
enable mod_rewrite support?" The purpose of this option is to get rid of
the index.php in the URL, resulting in a cleaner URL structure. In your
installation, you probably have this option enabled. In that case, it would
generate the URL's without the index.php, instead relying on the rewrite
rules to load index.php without displaying it in the URL.

As I mentioned, none of my other plugins have any routing issues, so I'm
pretty sure there's nothing wrong with OpenVBX itself. I think you must be
doing something in your code that doesn't follow OpenVBX standards,
probably hard-coding something that you should be pulling from a variable.
ᐧ

**Comment by jpwalters** [comment]
I just re-coded how the plugin.json file is accessed and this should resolve the issue you are seeing.  I'll post a version 1.3 shortly with the fix.

**Comment by jpwalters** [comment]
This was fixed in version v1.3 - https://github.com/jpwalters/OpenVBX-Plugin-SMS-Conversation/releases/tag/v1.3

**Comment by bkonia** [comment]
That fixed the issue, thanks!
ᐧ
